internal/kmsg/kmsg.go

```diff
@@ -0,0 +1,120 @@
+// Package kmsg contains support for parsing Linux kernel log entries read from
+// /dev/kmsg. These are the same log entries that can be read via the `dmesg`
+// command. Each read from /dev/kmsg is guaranteed to return a single log entry,
+// so no line-splitting is required. If the read buffer is not large enough to
+// contain the log entry, `os.ErrInvalid` is returned.
+//
+// More information can be found here:
+// https://www.kernel.org/doc/Documentation/ABI/testing/dev-kmsg
+package kmsg
+
+import (
+	"errors"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	// ErrInvalidFormat indicates the kmsg entry failed to parse.
+	ErrInvalidFormat = errors.New("invalid kmsg format")
+)
+
+// Entry is a single log entry in kmsg.
+type Entry struct {
+	Priority           uint8
```

Comment:
It looks like kmsg uses the standard syslog priority levels. I think we should define a type with those values, and use that instead of a raw uint8. I would use the keywords here for the names: https://en.wikipedia.org/wiki/Syslog#Severity_level

Reply:
I mostly care about this now because we allow it as a filtering parameter, and it's a lot more meaningful to see that we are filtering to "<= info" rather than "<= 4".

Reply:
I agree. An enum would be easier to reason with

Reply:
@kevpar You think logging the priority name would be a good idea as well? I don't think it has to be a member of the struct but specifically referring to adding a String method to the LogLevel type I'm going to make and doing this

Reply:
@kevpar You think we should also log the priority name 'Warning', 'Emergency' etc with the other fields? I don't think it has to be a member of the struct but specifically referring to adding a String method to the **LogLevel** type I'm going to make and just adding it as a field on the **logrus.Fields** thats returned from **logFormat**. 

```go
// Entry is a single log entry in kmsg.
type Entry struct {
	Priority           LogLevel
	Facility           uint8
	Seq                uint64
	TimeSinceBootMicro uint64
	Flags              string
	Message            string
}

func (ke *Entry) logFormat() logrus.Fields {
	return logrus.Fields{
		"priority":           ke.Priority,
		"priority-name":      ke.Priority.String(),
		"facility":           ke.Facility,
		"seq":                ke.Seq,
		"timesincebootmicro": ke.TimeSinceBootMicro,
		"flags":              ke.Flags,
		"message":            ke.Message,
	}
}
```

Reply:
I'd be fine with logging the priority name instead of the integer, but I don't think it's worth logging both.